src/browser/screen.js

```diff
@@ -207,12 +207,12 @@ function ScreenAdapter(screen_container, bus)
                 }
             }
 
-            if(cursor_element.style.display !== "none")
+            if(cursor_element.style.display !== "none" && cursor_row < text_mode_height && cursor_col < text_mode_width)
             {
                 context.fillStyle = cursor_element.style.backgroundColor;
                 context.fillRect(
                     cursor_col * char_size[0],
-                    cursor_row * char_size[1] + parseInt(cursor_element.style.marginTop, 10) - 1,
+                    cursor_row * char_size[1] + parseInt(cursor_element.style.marginTop, 10),
```

Comment:
-1 removed. 
if cursor_row is 0 and marginTop is 0 then fillRect should start with 0, not -1